[storage-file-share] README files for samples directories (#6208)
Azure/azure-sdk-for-js: sdk/storage/storage-file-share/samples/javascript/README.md

---
page_type: sample
languages:
- javascript
products:
- azure
- azure-storage

> urlFragment: storage-file-share-javascript

Author: @kaerm my understanding is that this `urlFragment` directive here will make it so the sample URL ends up being pinned to docs.microsoft.com/samples/Azure/azure-sdk-for-js/storage-file-share-javascript , right? Is there a preferred naming scheme for this?

Reviewer: that's how I understand it as well

Reviewer 2: Is there any preview link for the generted docs?

Reviewer: i do not think so. Once we have it merged we can see how it generates, and if change is needed it will be automatically picked up

---

# Azure Storage File Share client library samples for JavaScript

These sample programs show how to use the JavaScript client libraries for Azure Storage File Shares in some common scenarios.

|__File Name__|__Description__|
|-------------|---------------|

> |[basic.js][basic]|authenticate with the service using an account name & key (or anonymously with a SAS URL), upload a file, list files and directories, and download a file to a string|

Reviewer 3: English nit: I feel like the descriptions should be full sentences, starting with a capital letter and ending with a period

Author: I originally wrote them as sentences and found that they all began with "This sample" and then had all the current words except in a different tense. So, I decided to change them to make them feel less repetitive when scanning top-to-bottom. If you feel strongly that they should be full sentences, then I can try to write better/more diversified sentences, or I can change them back, but I think it subtracts from the information in the table.

|[withConnString.js][withConnString]|connect to and authenticate with the service using a connection string|
|[sharedKeyCred.js][sharedKeyCred]|authenticate with the service using an account name and a shared key|
|[anonymousCred.js][anonymousCred]|authenticate with the service anonymously using a SAS URL|
|[proxyAuth.js][proxyAuth]|connect to the service using a proxy and authenticate with an account name & key|
|[iterators-shares.js][iterators-shares]|connect to the service and iterate through the shares in the account|
|[iterators-files-and-directories.js][iterators-files-and-directories]|create a few directories and iterate through them individually (using async `for await` syntax), by page, and resuming paging using a marker|
|[iterators-handles.js][iterators-handles]|connect to the service and iterate through open handles|
|[customPipeline.js][customPipeline]|use custom HTTP pipeline options when connecting to the service|
|[advanced.js][advanced]|use custom logging and pipeline options, then upload a local file to a share|

## Prerequisites

> The sample code is compatible with Node.js. The samples are compatible with Node.js >= 8.0.0, except the samples that use the async `for await` syntax. Samples that use the async `for await` syntax require a version of Node that is compatible with that syntax (>= 10.0.0).

Reviewer 3: ```suggestion
Samples are compatible with Node.js >= 8.0.0, except for samples which use `for await` syntax, which require Node >= 10.0.0.
```

> You need [an Azure subscription][freesub] and [an Azure Storage account][azstorage] to run these sample programs. Samples retrieve credentials to access the storage account from environment variables. See each individual sample for details on which environment variables it requires to function.

Reviewer 2: Need to fillin actual links here.

Adapting the samples to run in the browser requires some additional consideration. For details, please see the [package README][package].

## Setup

To run the samples using the published version of the package:

1. Install the client library using `npm`:
```bash
npm install @azure/storage-file-share
```

> 2. Copy and modify the sample code, replacing the relative `require('../../')` statements with `require('@azure/storage-file-share')`)

Author: It's really unfortunate that one of the steps has to be "edit this code", since we are using relative imports in these files, which won't work for the samples after they are zipped up. The only alternative to editing the files is to clone the whole repo and build the client sdk using Rush, which seems even more unfortunate for the end-user of the samples.

Reviewer 4: Ideally, we shouldn't have this step. I'd rather have `require('@azure/storage-file-share')` checked in to git and we change it to `require('../..')` before trying to run it.

Something similar is what I tried for service bus at https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/servicebus/service-bus/package.json#L44

Author: I like the idea of pre-processing them fro CI and checking in the camera-ready copies. I would want to move the script to a shared location, though. I think we probably also want package.json (and tsconfig for the typescript ones) for the samples, that way the user experience on docs.microsoft.com is just download/extract the zip, run `npm install` (`npm run build` as well for ts, and we don't have to instruct the user to globally install TypeScript), then run a sample in Node. It would be awesome for these to be as turnkey as possible.

It feels to me like a lot of this process could be automated using a script to generate such files from a template. If we put good doc comments at the top of every sample explaining them, then we can use that text to generate the table of files in the readme and we could generate the title from the package name, the one-line description from package metadata, etc.

Reviewer 2: Vote for the idea : )

Reviewer 4: +1 for the automation, lets log an issue to track that.
In this PR, lets do the what's needed to have the samples light up in the docs
